spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-nacos-config/src/main/java/com/alibaba/cloud/nacos/configdata/NacosConfigDataLocationResolver.java

```diff
@@ -0,0 +1,250 @@
+		}
+
+		return nacosConfigProperties;
+	}
+
+	private BindHandler getBindHandler(ConfigDataLocationResolverContext context) {
+		return context.getBootstrapContext().getOrElse(BindHandler.class, null);
+	}
+
+	protected Log getLog() {
+		return this.log;
+	}
+
+	@Override
+	public boolean isResolvable(ConfigDataLocationResolverContext context,
+			ConfigDataLocation location) {
+		if (!location.hasPrefix(getPrefix())) {
+			return false;
+		}
+		return context.getBinder()
+				.bind(NacosConfigProperties.PREFIX + ".enabled", Boolean.class)
+				.orElse(true);
+	}
+
+	protected String getPrefix() {
+		return PREFIX;
+	}
+
+	@Override
+	public List<NacosConfigDataResource> resolve(
+			ConfigDataLocationResolverContext context, ConfigDataLocation location)
+			throws ConfigDataLocationNotFoundException,
+			ConfigDataResourceNotFoundException {
+		return Collections.emptyList();
+	}
+
+	@Override
+	public List<NacosConfigDataResource> resolveProfileSpecific(
+			ConfigDataLocationResolverContext resolverContext,
+			ConfigDataLocation location, Profiles profiles)
+			throws ConfigDataLocationNotFoundException {
+		NacosConfigProperties properties = loadProperties(resolverContext);
+
+		ConfigurableBootstrapContext bootstrapContext = resolverContext
+				.getBootstrapContext();
+
+		bootstrapContext.registerIfAbsent(NacosConfigProperties.class,
+				InstanceSupplier.of(properties));
+
+		registerConfigService(properties, bootstrapContext);
+
+		return loadConfigDataResources(location, profiles, properties);
+	}
+
+	private List<NacosConfigDataResource> loadConfigDataResources(
+			ConfigDataLocation location, Profiles profiles,
+			NacosConfigProperties properties) {
+		List<NacosConfigDataResource> result = new ArrayList<>();
+		String uris = getUri(location, properties);
+
+		if (StringUtils.isBlank(dataIdFor(uris))) {
+			throw new IllegalArgumentException("dataId must be specified");
+		}
+
+		NacosConfigDataResource resource = new NacosConfigDataResource(properties,
+				location.isOptional(), profiles, log,
+				new NacosItemConfig().setGroup(groupFor(uris, properties))
+						.setDataId(dataIdFor(uris)).setSuffix(suffixFor(uris, properties))
+						.setRefreshEnabled(refreshEnabledFor(uris, properties)));
+		result.add(resource);
+
+		return result;
+	}
+
+	private String getUri(ConfigDataLocation location, NacosConfigProperties properties) {
+		String path = location.getNonPrefixedValue(getPrefix());
+		if (StringUtils.isBlank(path)) {
+			path = "/";
+		}
+		if (!path.startsWith("/")) {
+			path = "/" + path;
+		}
+		return properties.getServerAddr() + path;
+	}
+
+	private void registerConfigService(NacosConfigProperties properties,
+									   ConfigurableBootstrapContext bootstrapContext) {
+		if (!bootstrapContext.isRegistered(ConfigService.class)) {
+			Optional.ofNullable(new NacosConfigManager(properties).getConfigService())
+					.ifPresent(configService -> bootstrapContext.register(
```

Comment:
not `new NacosConfigManager` to get `ConfigService`. 
I think you should register The `NacosConfigManager` with the `ConfigurableBootstrapContext ` and then use the `NacosConfigManager#getConfigService()`

Reply:
Fixed